### PR TITLE
Change `--font-weight-bold` to `--font-weight-semibold` and 600 value, introduce new font weight variables

### PR DESCRIPTION
**Pull request by silverwind**
There was some recent discussion about this in Discord `ui-design` channel and the conclusion was that https://github.com/go-gitea/gitea/issues/24305 should have fixed their OS font installation to have semibold weights.

I have now tested this 601 weight on a Windows 10 machine on Firefox myself, and I immediately noticed that bold was excessivly bold and rendering as 700 because browsers are biased towards bolder fonts. So revert this back to the previous value.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Comment could be left there (and be improved), then some Linux users without correct font environment could know the details, and it could save contributors time if similar issue comes.

**Comment by silverwind** [comment]
Hmm, I don't feel like commenting this, 600 is a pretty standard weight. Backport will be done, yes.

**Comment by wxiaoguang** [comment]
We have spent a lot of time on this problem.

Maybe in the future some other users would also report similar problems, then some new maintainers without this knowledge would waste time on this problem again.

**Comment by wxiaoguang** [comment]
Well, a new issue ...   Bold text no longer displayed as bold #24835

**Comment by silverwind** [comment]
> We have spent a lot of time on this problem.
> 
> Maybe in the future some other users would also report similar problems, then some new maintainers without this knowledge would waste time on this problem again.

What should we write? 601 was a "odd" value that caused issues and renders as 700 which we do not want from a design perspective. Multiples of 100 seem to generally work best.

For clarity, I would like to rename the variable. As per MDN:

400 | Normal
500 | Medium
600 | Semi Bold
700 | Bold

So we could make:

--font-weight-normal
--font-weight-medium
--font-weight-semibold
--font-weight-bold

That will satisfy all future use cases.

**Comment by wxiaoguang** [comment]
I think the "situation" could be documented:

```
// "bold" can't be used because it's too bold for most cases
// some fonts (eg: Segoe UI on Linux) do not provide "bolding" for weight 600, end users should fix their font library
```

The key information:

1. Why not use "bold", it has been asked not just one time, I also asked similar question before, the latest question see https://github.com/go-gitea/gitea/issues/24835#issuecomment-1556222307
2. If any user doesn't see "bold" font, that's their system's problem, not Gitea's problem.

It would save a lot of time for future developers.

File: web_src/css/base.css [hunk 1]
```diff
@@ -3,7 +3,11 @@
   --fonts-proportional: -apple-system, "Segoe UI", system-ui, Roboto, "Helvetica Neue", Arial;
   --fonts-monospace: ui-monospace, SFMono-Regular, "SF Mono", Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace, var(--fonts-emoji);
   --fonts-emoji: "Apple Color Emoji", "Segoe UI Emoji", "Noto Color Emoji", "Twemoji Mozilla";
-  --font-weight-bold: 600;
+  /* font weights */
+  --font-weight-normal: 400;
+  --font-weight-medium: 500;
+  --font-weight-semibold: 600;
+  --font-weight-bold: 700;
```
Comment (wxiaoguang) [inline web_src/css/base.css]: I guess unused variables should be removed?

**Review by wxiaoguang — APPROVED** [review]
If you wouldn't like to write comment or remove unused variable, I could do it in next PR, I won't block for nits.

**Comment by silverwind** [comment]
Will do a bit more refactor to use variables everywhere.

File: web_src/css/base.css (hunk 1 above)
Comment (silverwind) [reply web_src/css/base.css]: Removed `--font-weight-bold` as I think it should be never used.

**Comment by silverwind** [comment]
Refactor done, now all CSS uses these variables and there are tailwind-matching helper classes as well for them.

**Comment by silverwind** [comment]
Oh, and there is one more change. Issue title went from 300 to 400, which matches GH:

Before:
Screenshot 2023-05-21 at 22 07 10

After:
Screenshot 2023-05-21 at 22 07 21

I think we should always use weights between 400 and 600, but I have kept the 300 weights for japanese fonts for now as I can not really judge those accurately.

**Comment by belzecue** [comment]
Some follow-up, as I was the author of #24305.

> There was some recent discussion about this in Discord ui-design channel and the conclusion was that https://github.com/go-gitea/gitea/issues/24305 should have fixed their OS font installation to have semibold weights.

Same wiki page on try.gitea.io, now working as expected -- with no font changes required at my end.

-----

Screenshot at 2023-07-06 13-48-42

-----

Details:

Screenshot at 2023-07-06 13-50-53

try.gitea.io is running 1.21, but I also upgraded my gitea cloud server to 1.19.4 and confirmed wiki page `**text**` markdown is bolding as expected.

Great outcome, thanks!

**Comment by belzecue** [comment]
> There was some recent discussion about this in Discord ui-design channel and the conclusion was that https://github.com/go-gitea/gitea/issues/24305 should have fixed their OS font installation to have semibold weights.

This is correct.  I am the submitter of #24305.  On the still affected device (MX Linux), i just now downloaded a Linux version of Segoe-UI from [here](https://github.com/mrbvrz/segoe-ui-linux), installed it and restarted my browser, and this device is now displaying the bold/semibold text correctly.

**Pull request merged**